pkg/planner/core/indexmerge_path.go

```diff
@@ -744,77 +554,44 @@ func generateIndexMerge4NormalIndex(ds *logicalop.DataSource, regularPathCount i
 	}
-	return "", nil
-}
-
-// generateIndexMergeOnDNF4MVIndex generates IndexMerge paths for MVIndex upon DNF filters.
-/*
-	select * from t where ((1 member of (a) and b=1) or (2 member of (a) and b=2)) and (c > 10)
-		IndexMerge(OR)
-			IndexRangeScan(a, b, [1 1, 1 1])
-			IndexRangeScan(a, b, [2 2, 2 2])
-			Selection(c > 10)
-				TableRowIdScan(t)
-	Two limitations now:
-	1). all filters in the DNF have to be used as access-filters: ((1 member of (a)) or (2 member of (a)) or b > 10) cannot be used to access the MVIndex.
-	2). cannot support json_contains: (json_contains(a, '[1, 2]') or json_contains(a, '[3, 4]')) is not supported since a single IndexMerge cannot represent this SQL.
-*/
-func generateIndexMergeOnDNF4MVIndex(ds *logicalop.DataSource, normalPathCnt int, filters []expression.Expression) (mvIndexPaths []*util.AccessPath, err error) {
-	for idx := 0; idx < normalPathCnt; idx++ {
-		if !isMVIndexPath(ds.PossibleAccessPaths[idx]) {
-			continue // not a MVIndex path
-		}
 
-		// for single MV index usage, if specified use the specified one, if not, all can be access and chosen by cost model.
-		if !isInIndexMergeHints(ds, ds.PossibleAccessPaths[idx].Index.Name.L) {
-			continue
-		}
+	if needConsiderIndexMerge {
+		return "", nil
+	}
 
-		for current, filter := range filters {
-			sf, ok := filter.(*expression.ScalarFunction)
-			if !ok || sf.FuncName.L != ast.LogicOr {
-				continue
+	var containMVPath bool
+	for i := regularPathCount; i < len(ds.PossibleAccessPaths); i++ {
+		path := ds.PossibleAccessPaths[i]
+		for _, p := range util.SliceRecursiveFlattenIter[*util.AccessPath](path.PartialAlternativeIndexPaths) {
+			if isMVIndexPath(p) {
+				containMVPath = true
+				break
 			}
-			dnfFilters := expression.SplitDNFItems(sf) // [(1 member of (a) and b=1), (2 member of (a) and b=2)]
+		}
 
-			unfinishedIndexMergePath := genUnfinishedPathFromORList(
-				ds,
-				dnfFilters,
-				[]*util.AccessPath{ds.PossibleAccessPaths[idx]},
-			)
-			finishedIndexMergePath := handleTopLevelANDList(
-				ds,
-				filters,
-				current,
-				[]*util.AccessPath{ds.PossibleAccessPaths[idx]},
-				unfinishedIndexMergePath,
-			)
-			if finishedIndexMergePath != nil {
-				mvIndexPaths = append(mvIndexPaths, finishedIndexMergePath)
-			}
+		if !containMVPath {
+			ds.PossibleAccessPaths = slices.Delete(ds.PossibleAccessPaths, i, i+1)
 		}
 	}
-	return
+	if len(ds.PossibleAccessPaths) == regularPathCount {
+		return "IndexMerge is inapplicable or disabled. ", nil
+	}
+	return "", nil
 }
 
-// generateIndexMerge4ComposedIndex generates index path composed of multi indexes including multivalued index from
-// (json_member_of / json_overlaps / json_contains) and single-valued index from normal indexes.
+// generateANDIndexMerge4ComposedIndex tries to generate AND type index merge AccessPath for (
+//json_member_of / json_overlaps / json_contains) on multiple multi-valued or normal indexes.
```

Comment:
Is this formatted by the `go fmt`? I thought it would always keep a space here.

Reply:
Updated.
This is formatted by GoLand's "wrap on typing", which is not very clever sometimes. It won't add a space in such places.